app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbFragment.java

```diff
@@ -35,20 +37,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         v = inflater.inflate(R.layout.breadcrumb_fragment, container, false);
         breadCrumb = (BreadcrumbView) v.findViewById(R.id.bread_bar);
+        if (savedInstanceState != null) {
+            currentPath = savedInstanceState.getString("currentPath");
+            breadCrumb.rootPath = savedInstanceState.getString("rootPath");
+            breadCrumb.highlightedItem = savedInstanceState.getString("highlightedItem");
+            breadCrumb.highlightedIndex= savedInstanceState.getInt("highlightedIndex");
+        }
         return v;
     }
 
     @Override
+    public void onStart() {
+        super.onStart();
+        breadCrumb = (BreadcrumbView) v.findViewById(R.id.bread_bar);
+        if (currentPath.length() > 0 && breadCrumb.rootPath .length() > 0) {
```

Comment:
This one is not formatted. Interestingly, `irootPath. length()` can pass compling.